### PR TITLE
PersistentFSM.StateChangeEvent extractor

### LINKED ISSUE #23072: cannot access PersistentFSM.StateChangeEvent from PersistentQuery streams
i would expect to be able to do something like:
```scala
PersistenceQuery(system)
        .readJournalFor[ScalaDslMongoReadJournal](MongoReadJournal.Identifier)
        .eventsByPersistenceId("id", 0, Long.MaxValue)
        .collect[String] {
          case StateChangeEvent(name, _) => name
        }
```
however, since StateChangeEvent is private to the persistence package (and rightly so), this can't be done.

maybe an extractor could be useful here?

### PR DESCRIPTION
**Pull request by mizerlou**
created extractor to pull the state name out of StateChangeEvent fixes #23072

### CODE REVIEW

File: akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala [hunk 1]
```diff
@@ -156,6 +156,16 @@ object PersistentFSM {
   private[persistence] case class StateChangeEvent(stateIdentifier: String, timeout: Option[FiniteDuration]) extends PersistentFsmEvent
 
   /**
+    * An extractor to access the name of the StateChangeEvent
+    */
+  object StateChangeEventExtractor {
```
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala]: shouldn't we just make the `StateChangeEvent` public instead?

Comment (mizerlou) [inline akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala]: sure.  i just figured you were hiding implementation details.

**Comment by mizerlou** [comment]
@patriknw i suppose this is ready to go now...

File: akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala [hunk 2]
```diff
@@ -152,8 +152,7 @@ object PersistentFSM {
    * @param stateIdentifier FSM state identifier
    * @param timeout FSM state timeout
    */
-  @InternalApi
-  private[persistence] case class StateChangeEvent(stateIdentifier: String, timeout: Option[FiniteDuration]) extends PersistentFsmEvent
+  case class StateChangeEvent(stateIdentifier: String, timeout: Option[FiniteDuration]) extends PersistentFsmEvent
```
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala]: `final` would be nice here

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
OK TO TEST

**Pull request merged**